### ISSUE TITLE
Upgrade to pydantic v2

### ISSUE THREAD

**Issue by pkdash** [I-GUIDE/catalog#38]
pydantic v2 provides performance improvements.

**Comment by horsburgh** [comment]
@pkdash - is this work complete?

**Comment by pkdash** [comment]
@horsburgh No. Lot of code has changed since I last worked on this upgrade. I have to bring this branch up to date with develop and make further code changes to meet Pydantic v2 requirements.
